source/common/request_info/request_info_impl.h

```diff
@@ -197,13 +203,15 @@ struct RequestInfoImpl : public RequestInfo {
   bool hc_request_{};
   const Router::RouteEntry* route_entry_{};
   envoy::api::v2::core::Metadata metadata_{};
+  absl::string_view requestedServerName_;
 
 private:
   uint64_t bytes_received_{};
   uint64_t bytes_sent_{};
   Network::Address::InstanceConstSharedPtr upstream_local_address_;
   Network::Address::InstanceConstSharedPtr downstream_local_address_;
   Network::Address::InstanceConstSharedPtr downstream_remote_address_;
+  absl::string_view requested_server_name_;
```

Comment:
I was wondering this as well. I just noted some of the lifetime best-practices [here](https://abseil.io/tips/1).  In the access log case, maybe  it would probably be safest to store as std::string or still accept string_view but immediately convert with string(string_view).

I only used absl::string_view since we pass it in [include/envoy/network/listen_socket.h](https://github.com/envoyproxy/envoy/blob/master/include/envoy/network/listen_socket.h#L164-L173)  and [include/envoy/connection.h](https://github.com/envoyproxy/envoy/blob/master/include/envoy/network/connection.h#L189-L192).

Reply:
I think it's safe, provided 2 things:

1) the name is only set once for a listener socket/connection and never changed. That implies TLS must not be renegotiated mid-connection. I don't know enough about TLS to know if that's true or not.

2) AccessLog::Instance implementations cannot keep the RequestInfoImpl object beyond the lifetime of the listener, which I think means beyond the call to `log` -- e.g. if they buffer log data it must have already been converted into a another format. The file and grpc loggers do their conversion in log. The `route_entry_` field also requires this behavior, so we're not worse off here.

I think in the end the fact that it's hard to reason about whether that string_view is still valid means we should copy it just to be safe unless there's a known performance implication.

Reply:
Did you decide there's a performance implication and want to keep the string view?

Reply:
cc @htuch @PiotrSikora since they may be more familiar with `absl::string_view` than I and may have more guidelines for it's lifetime quirks

Reply:
Yeah, should be safe, but given how much other stuff is in `RequestInfoImpl`, a copy is probably fast enough and safe; e.g. maybe you server name is only 12 bytes on average, that's like 1.5 pointers worth of space/copy overhead.